include/swift/AST/DiagnosticsParse.def

```diff
@@ -131,17 +131,17 @@ ERROR(lex_unicode_escape_braces,none,
 ERROR(lex_illegal_multiline_string_start,none,
       "multi-line string literal content must begin on a new line", ())
 ERROR(lex_illegal_multiline_string_end,none,
-      "multi-line string literal content must not be on the delimiter's line", ())
-ERROR(lex_insufficient_string_indent,none,
-      "string literal content is insufficiently indented", ())
-ERROR(lex_unexpected_tab_in_string_indent,none,
-      "string literal content is indented with a tab where a space is expected", ())
-ERROR(lex_unexpected_space_in_string_indent,none,
-      "string literal content is indented with a space where a tab is expected", ())
+      "multi-line string literal closing delimiter must begin on a new line", ())
+ERROR(lex_inconsistent_string_indent,none,
+      "line does not start with same indentation as closing delimiter of multi-line string literal", ())
+NOTE(note_insufficient_string_indent,none,
+      "unexpectedly found non-indentation character here", ())
+NOTE(note_unexpected_char_in_string_indent,none,
+      "unexpected %select{space|tab}0 in indentation here", (bool))
+NOTE(note_matching_indentation_char_here,none,
+      "should match %select{space|tab}0 in same position before closing delimiter", (bool))
```

Comment:
Counterpoint: https://bugs.swift.org/browse/SR-4715. When the user makes a mistake, they want to fix it completely without having to recompile. That at least means one error with multiple notes.

(I really don't trust a fix-it for the whole thing because of content indentation:

```
func foo() {
  let defs = """
    (define (cdar x)
      (cdr (car x)) )
    (define (cadr x)
<TAB>  (car (cdr x)) )
    """
  print(defs)
}
```

)